auth-svc/controllers/user-controller.js

```diff
@@ -1,22 +1,40 @@
+const UserRepository = require('../service/db-service/UserRepository');
 const userService = require('../service/user-service');
-const { validationResult } = require('express-validator');
+const mailService = require('../service/mail-service');
+const tokenService = require('../service/token-service');
 const ApiError = require('../exeptions/api-error');
+const UserDto = require('../dtos/user-dto');
 
 class UserController {
   async registration(req, res, next) {
     try {
-      const errors = validationResult(req);
-      if (!errors.isEmpty()) {
-        return next(ApiError.BadRequest('Validation error', errors.array()));
+      const { email } = req.body;
+      const userExists = await UserRepository.findByEmail(email);
+
+      if (userExists) {
+        throw ApiError.BadRequest(`User with email ${email} already exists`);
       }
-      const { email, password, name, last_name, role } = req.body;
-      const userData = await userService.registration(
+
+      const registrationPayload =
+        await userService.buildUserRegistrationPayload(req.body);
+      const user = await UserRepository.createUser(registrationPayload);
+
+      console.log(user);
```

Comment:
remove console log statement if you're done testing.

Reply:
Lol, i'm sorry, i don't know why it's here